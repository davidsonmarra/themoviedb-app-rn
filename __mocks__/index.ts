import MovieDTO from '../src/@types/movies-dto';

export const mockCardMovie: MovieDTO = {
  id: 1,
  title: 'Movie Title',
  overview: 'Movie Overview',
  poster_path:
    'https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg',
  release_date: '2021-01-01',
  backdrop_path:
    'https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg',
  vote_average: 10,
  genre_ids: [1, 2],
};

export const mockListMovie: MovieDTO[] = [
  {
    id: 1,
    title: 'Movie Title',
    overview: 'Movie Overview',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 10,
    genre_ids: [1, 2],
  },
  {
    id: 2,
    title: 'Movie Title 2',
    overview: 'Movie Overview 2',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 9,
    genre_ids: [1, 2],
  },
];
