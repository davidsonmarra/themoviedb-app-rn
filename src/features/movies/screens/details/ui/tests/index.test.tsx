import React from 'react';
import DetailsContainer from '..';
import {render, screen} from '@testing-library/react-native';
import {mockCardMovie} from '../../../../../../../__mocks__';

const mockGoBack = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({goBack: mockGoBack}),
}));

const containerInstance = (props: any) =>
  render(<DetailsContainer {...props} />);

describe('DetailsContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({
      movie: mockCardMovie,
      genresMovie: [{id: 1, name: 'Action'}],
    }).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should render movie details', () => {
    containerInstance({
      movie: mockCardMovie,
      genresMovie: [{id: 1, name: 'Action'}],
    });
    const title = screen.getByText(mockCardMovie.title);
    expect(title).toBeTruthy();
  });
});
