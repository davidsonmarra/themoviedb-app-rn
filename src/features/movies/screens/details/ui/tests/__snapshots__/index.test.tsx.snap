// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsContainer should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#01B4A4",
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
      testID="back-button"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="#FFFFFF"
        focusable={false}
        height="24"
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={960}
        vbWidth={960}
        width="24"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M359-242L120-481l239-239 43 43-166 166h604v60H236l166 166-43 43z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          {
            "flex": 1,
            "textAlign": "center",
          },
        ]
      }
    >
      Detalhes
    </Text>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
      testID="right-buton-icon"
    />
  </View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "backgroundColor": "rgba(0,0,0,0.75)",
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 24,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#121212",
              "fontSize": 24,
              "fontWeight": "bold",
            },
            {
              "color": "#FFFFFF",
              "marginBottom": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Movie Title
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "marginBottom": 24,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 16,
              },
              undefined,
            ]
          }
        >
          31/12/2020
           â€¢
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                },
                undefined,
              ]
            }
          >
             
            Action
          </Text>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
            },
            {
              "marginBottom": 8,
            },
          ]
        }
      >
        Sinopse
      </Text>
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 16,
            },
            undefined,
          ]
        }
      >
        Movie Overview
      </Text>
      <View
        style={
          {
            "marginHorizontal": "auto",
            "marginTop": 24,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderColor": "#00C851",
              "borderRadius": 32,
              "borderWidth": 5,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "width": 64,
            }
          }
          testID="average-container"
        >
          <Text
            style={
              [
                {
                  "color": "#121212",
                  "fontSize": 24,
                  "fontWeight": "bold",
                },
                {
                  "color": "#00C851",
                  "fontSize": 28,
                },
              ]
            }
            testID="average-text"
          >
            10.0
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
